example/select/select_framehandle_test.c

```diff
@@ -0,0 +1,314 @@
+
+      zVec3D dir_3D;
+      zVec3DMul( &g_parts_axis_unit_3D_vector, project_3D_scale, &dir_3D );
+
+      /* translate frame */
+      zVec3DAdd( &g_frame3D_org.pos, &dir_3D, &g_fh.frame.pos );
+
+      /* end of translate mode */
+    } else{
+      /* rotate mode */
+      zVec3D vp_radial_dir_center_to_goal;
+      zVec3DSub( &vp_mouse_goal_3D, &g_vp_circle_center_3D, &vp_radial_dir_center_to_goal);
+
+      if ( !zVec3DIsTiny( &g_vp_radial_dir_center_to_start ) ){
+        /* oval circle rotation on view port plane */
+        double theta = zVec3DAngle( &g_vp_radial_dir_center_to_start, &vp_radial_dir_center_to_goal, &g_parts_axis_unit_3D_vector);
+        zMat3D m;
+        /* zVec3D axis; */
+        if( selected_parts_id == 3 ){
+          /* x */
+          zMat3DFromZYX( &m, 0, 0, theta );
+        } else if( selected_parts_id == 4 ){
+          /* y */
+          zMat3DFromZYX( &m, 0, theta, 0 );
+        } else if( selected_parts_id == 5 ){
+          /* z */
+          zMat3DFromZYX( &m, theta, 0, 0 );
+        } else{
+          ZRUNERROR( "selected_parts_id is ERROR\n" );
+        }
+        zMulMat3DMat3D( &g_frame3D_org.att, &m, &g_fh.frame.att );
+      }
+      /* end of if g_vp_radial_dir_center_to_start is not Tiny */
+    }
+    /* end of rotate mode */
+  }
+  /* end moving mode */
+}
+
+void resize(int w, int h)
+{
+  rkglVPCreate( &cam, 0, 0, w, h );
+  rkglOrthoScale( &cam, g_scale, g_znear, g_zfar );
+}
+
+void keyboard(unsigned char key, int x, int y)
+{
+  switch( key ){
+  case 'u': rkglCALockonPTR( &cam, 5, 0, 0 ); break;
+  case 'U': rkglCALockonPTR( &cam,-5, 0, 0 ); break;
+  case 'i': rkglCALockonPTR( &cam, 0, 5, 0 ); break;
+  case 'I': rkglCALockonPTR( &cam, 0,-5, 0 ); break;
+  case 'o': rkglCALockonPTR( &cam, 0, 0, 5 ); break;
+  case 'O': rkglCALockonPTR( &cam, 0, 0,-5 ); break;
+  case '8': g_scale += 0.001; rkglOrthoScale( &cam, g_scale, g_znear, g_zfar ); break;
+  case '*': g_scale -= 0.001; rkglOrthoScale( &cam, g_scale, g_znear, g_zfar ); break;
+  case '9': rkglCARelMove( &cam, 0, 0.05, 0 ); break;
+  case '(': rkglCARelMove( &cam, 0,-0.05, 0 ); break;
+  case '0': rkglCARelMove( &cam, 0, 0, 0.05 ); break;
+  case ')': rkglCARelMove( &cam, 0, 0,-0.05 ); break;
+  case 'q': case 'Q': case '\033':
+    exit( EXIT_SUCCESS );
+  default: ;
+  }
+}
+
+void init(void)
+{
+  rkglSetCallbackParamGLUT( &cam, 0, 0, 0, 0, 0 );
+
+  rkglBGSet( &cam, 0.5, 0.5, 0.5 );
+  rkglCASet( &cam, 5, 0, 2, 0, -20, 0 );
+
+  glEnable( GL_LIGHTING );
+  rkglLightCreate( &light, 0.4, 0.4, 0.4, 1, 1, 1, 0, 0, 0 );
+  rkglLightMove( &light, 8, 0, 8 );
+
+  int i;
+  for( i=0; i<NOBJECTS; i++ ){
+    g_fh.partsInfo[i].updown = 0;
+    glDeleteLists( g_fh.partsInfo[i].list, 1 );
+    g_fh.partsInfo[i].list = glGenLists( 1 );
+    glNewList( g_fh.partsInfo[i].list, GL_COMPILE );
+    glEndList();
```

Comment:
It is a mistake to not register anything here by glGenLists(), glNewList(), glEndList().

Reply:
I Fixed. The displayLists glGenLists(), glNewList(), glEndList(), and glCallLists() are enabled.